### PR TITLE
front: transform some RTK mutations into queries to prevent useless requests

### LINKED ISSUE #9830: front: useless infra requests are sent
### Description and goal

Sometimes, useless infra requests are fired.

For instance, you can open the network and check the requests sent after selecting a new train on the scenario interface. 

### Acceptance criteria

- [ ] no more useless requests

### PR DESCRIPTION
**Pull request by clarani**
closes #9830 

See the commits

### CODE REVIEW

File: front/src/applications/operationalStudies/views/SimulationResults.tsx [hunk 1]
```diff
@@ -158,33 +155,36 @@ const SimulationResults = ({
...
-            <SpeedSpaceChartContainer
-              trainSimulation={trainSimulation}
-              selectedTrainPowerRestrictions={selectedTrainPowerRestrictions}
-              rollingStock={selectedTrainRollingStock}
-              pathProperties={pathProperties}
-              heightOfSpeedSpaceChartContainer={speedSpaceChartContainerHeight}
-              setHeightOfSpeedSpaceChartContainer={setSpeedSpaceChartContainerHeight}
-            />
+      {selectedTrainRollingStock &&
+        trainSimulation?.status === 'success' &&
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/views/SimulationResults.tsx]: This should be enough because `useSimulationResults` returns only the success simulation (or undefined). Same for the map and output table.

```suggestion
        trainSimulation &&
```

File: front/src/applications/operationalStudies/views/SimulationResults.tsx [hunk 2]
```diff
@@ -95,10 +96,6 @@ const SimulationResults = ({
     }
   }, [extViewport]);
 
-  if (!trainSimulation) return null;
-
-  if (trainSimulation.status !== 'success') return null;
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/views/SimulationResults.tsx]: This doesn't seem to concern the move of `useSimulationResults` ? It fixes a bug ? This might be better to put it in a different commit. Same for `pathProperties` for the space time chart.

Comment (clarani) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: You're right :+1:

File: front/src/applications/operationalStudies/views/SimulationResults.tsx (hunk 1 above)
Comment (clarani) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: Done

File: front/src/common/api/osrdEditoastApi.ts [hunk 3]
```diff
@@ -32,16 +33,27 @@ const osrdEditoastApi = generatedEditoastApi.enhanceEndpoints({
...
+    // remove 'infra' from invalidatesTags of the following endpoints
+    postInfraByInfraIdObjectsAndObjectType: {
+      invalidatesTags: [],
+    },
+    postInfraByInfraIdPathfinding: {
+      invalidatesTags: [],
+    },
+    postInfraByInfraIdRoutesNodes: {
+      invalidatesTags: [],
+    },
```
Comment (emersion) [inline front/src/common/api/osrdEditoastApi.ts]: I wonder, should we convert these requests to queries instead of mutations?

Comment (SharglutDev) [reply front/src/common/api/osrdEditoastApi.ts]: I vote yes :) @kmer2016 @Math-R

File: front/src/applications/operationalStudies/views/SimulationResults.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: Thank you. Could you add a description in this new commit to explain why we don't make these checks anymore ?

File: front/src/applications/operationalStudies/views/SimulationResults.tsx (hunk 2 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: Thank you. Could you add a description in this new commit to explain why we don't make these checks anymore ?

Comment (clarani) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: I removed these commits and moved them in a new PR: https://github.com/OpenRailAssociation/osrd/pull/9885

Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: You didn't keep theses changes after all after the mutation > query refacto ?

Maybe you could also rename this PR and update the description :)

Edit : oups we posted at the same time :)

**Comment by SharglutDev** [comment]
Since your rebase and the merge of this [PR](https://github.com/OpenRailAssociation/osrd/pull/9614), there is another spot where you need to update the `useMutation` in `useCachedTrackSections` :)

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, nice changes !

**Review by emersion — APPROVED** [review]
